src/hypothesis/strategies.py

```diff
@@ -542,6 +542,38 @@ def frozensets(elements=None, min_size=None, average_size=None, max_size=None):
 
 
 @defines_strategy
+def iterables(elements=None, min_size=None, average_size=None, max_size=None,
+              unique_by=None, unique=False):
+    """This has the same behaviour as lists, but returns iterables instead.
+
+    It is easy to forget that some iterables cannot be indexed (e.g. sets)
```

Comment:
I’m generally uncomfortable about the use of “easy” in docs – it makes assumptions about the reader’s technical level that are usually unhelpful. I don’t think this is as bad as some, but I think we could make it a little neater. And we say “cannot be indexed” and “do not support indexing” – what’s the difference?

How about:

> Some iterables cannot be indexed (e.g. sets) and some do have a fixed length (e.g. generators). This strategy produces iterators, which cannot be indexed and do not have a fixed length. This ensures you do not accidentally depend on sequence behaviour.

Also "produces" just to avoid confusion with "generators". 😄